File: server/src/main/java/br/com/fastgondolas/fastos/server/model/Endereco.java
package br.com.fastgondolas.fastos.server.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "enderecos")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Endereco extends BaseEntity {

    private String logradouro;
    private String numero;
    private String complemento;
    private String bairro;
    private String cidade;
    private String estado;
    private String cep;
    private String descricao;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;
}

File: mobile/src/db/models/Endereco.ts
import { Model, Relation } from '@nozbe/watermelondb';
import {
  field,
  relation,
  text,
  readonly,
  date,
} from '@nozbe/watermelondb/decorators';
import { Cliente } from './Cliente';

export default class Endereco extends Model {
  static override table = 'enderecos';

  static associations = {
    clientes: { type: 'belongs_to', key: 'cliente_id' },
  } as const;

  @text('logradouro') logradouro!: string;
  @field('numero') numero!: string;
  @field('cidade') cidade!: string;
  @field('estado') estado!: string;
  @field('cliente_id') clienteId!: string;

  @readonly @date('created_at') createdAt!: Date;
  @readonly @date('updated_at') updatedAt!: Date;

  @relation('clientes', 'cliente_id') cliente!: Relation<Cliente>;
}
